katdal/datasources.py

```diff
@@ -83,6 +84,21 @@ def shape(self):
         return self.vis.shape
 
 
+def _has_chunk_to_flags(has_chunk, block_id=None, full_chunks=None):
```

Comment:
These defaults should never be used and I hope things crash if they are. The function is only meant to be used together with `da.map_blocks`, which passes in the expected kwargs from its own kwargs.

I'm following the example straight from the `da.map_blocks` docstring which explains the special behaviour of the `block_id` keyword, if given. That default of `None` is also invalid but will never occur.

...

So I've checked: I can safely delete the defaults, so that's what I'll do.